[16.0] [ADD] cross_connect_client

paradoxxxzero opened the pull request:
This module allows an odoo instance users to connect directly on another odoo instance
where the module `cross_connect_server` (https://github.com/OCA/server-auth/pull/734) is installed.

paradoxxxzero committed:
[ADD] cross_connect_client

paradoxxxzero committed:
[IMP] cross_connect_client: Handle redirect_url in controller

sebastienbeau committed:
cross_connect_client: only pass redirect_url if set

sebastienbeau committed:
cross_connect_client: by default open in a new windows as we change of domain name

paradoxxxzero committed:
[IMP] cross_connect_client: Require and send mail

florian-dacosta commented on cross_connect_client/models/cross_connect_server.py:
```diff
@@ -0,0 +1,180 @@
+    _inherit = "server.env.mixin"
+
+    name = fields.Char(
+        required=True, compute="_compute_name", readonly=False, store=True
```
I would add a `help` to explain that this name will be the new "app" name.
Also, I wonder why it is computed from the `server_url`. 
I mean, it does not cost much to have a normal (not computed) required field, as it is quite important.
I don't see in which case we could want to have the new app with the server url name.

paradoxxxzero replied:
Hostname but it’s just for minor convenience, might not be needed, remove the attrs in the view in this case.

florian-dacosta replied:
what about the removal of the compute method ?

paradoxxxzero replied:
! meant also the attrs :sweat_smile:

florian-dacosta commented on cross_connect_client/models/cross_connect_server.py:
```diff
@@ -0,0 +1,180 @@
+
+    def action_disable(self):
+        for record in self:
+            record.group_ids.unlink()
```
I don't like this button very much because I think it is easy to click without knowing the consequences (deletion of the group). Because clicking on `action_disable`, then `action_sync` is not without consequence since you would have to reconfigure the group on all users needing support.

I wonder if we could do something like `record.menu_id.unlink()` instead.
action_sync re-create the menu anyway (or if it does not if not group is created, we could force it.

Then, the `action_disable` would disable the menu but would still keep all user / group configuration.
What do you think ?

paradoxxxzero replied:
Maybe archiving menu entries in this case?

florian-dacosta replied:
Well why not, we'll just have to manage the active state in the `_compute_menu_id` then.
But ok for me, I 'd just prefer avoiding the groups deletion

florian-dacosta commented on cross_connect_client/models/cross_connect_server.py:
```diff
@@ -0,0 +1,180 @@
+
+    @property
+    def _server_env_fields(self):
+        return {"api_key": {}}
```
I think we should override the `unlink` method to delete the groups and menu before deleting the cross connect server.
(Even without changing the `action_disable` method, I guess the `cross.connect.server` could be deleted while enabled)

paradoxxxzero replied:
Can’t we cascade these?

florian-dacosta replied:
Better indeed.

florian-dacosta commented on cross_connect_client/models/cross_connect_server.py:
```diff
@@ -0,0 +1,180 @@
+                record.menu_id = False
+                continue
+
+            menu_groups = self.env.ref("base.group_system") | record.group_ids
```
`self.env.ref("base.group_system")` should be removed IMO because, the group does not come from server so it will display the new app menu but won't allow you to access it.

paradoxxxzero replied:
Agreed

florian-dacosta reviewed (approved):
LGTM test + code review.
I'd added some not blocking remarks, but is good enough to be merged already IMO.

florian-dacosta committed:
[IMP] Remove default value for the name of the server

florian-dacosta committed:
[IMP] Avoid deleting the cross connect server groups when unactiving it
We want to avoid deletion of the group because we would then loose all users configured on it. So we prefer to remove the link between the cross connect server and the group instead

paradoxxxzero committed:
Merge pull request #3 from akretion/16.0-add-cross_connect_client-fdc-ref

[REF] avoid groupe deletion + simplify name management

sebastienbeau reviewed (approved):
LGTM (code review)

sebastienbeau commented:
/ocabot merge nobump
